tests/integration/targets/nxos_vrf_interfaces/tests/common/_gather_interfaces.yaml

```diff
@@ -0,0 +1,10 @@
+---
+- name: Gather the provided configuration
+  register: result
+  cisco.nxos.nxos_vrf_interfaces:
```

Comment:
Why are we using nxos_vrf_interfaces in tests for the same module?

Reply:
this is only to gather the before state to asset it, hence using this

Reply:
This is not a good approach, IMO. I know we've done it in a couple of places, but we should actively avoid it if there's an alternative.

If the `state: gathered` is broken (has incorrect logic), then the entire premise of testing this module becomes invalid.

Reply:
i can do the same for asserting the before state as well, ill check if we are having a specified before state for those N interfaces we touch